expression/builtin_encryption_vec.go

```diff
@@ -599,7 +600,12 @@ func (b *builtinCompressSig) vecEvalString(input *chunk.Chunk, result *chunk.Col
 			result.AppendString("")
 		}
 
-		compressed, err := deflate(hack.Slice(str))
+		str, err = bufEnc.EncodeString(str)
```

Comment:
It seems that `EncodeString` don't use a existing buffer, and `deflate` don't modify the slice, so IMHO we can use `buf.GetBytes` to replace `buf.GetString`. I tested on my local env, but I'm not so sure....

Reply:
> It seems that `EncodeString` don't use a existing buffer, and `deflate` don't modify the slice, so IMHO we can use `buf.GetBytes` to replace `buf.GetString`. I tested on my local env, but I'm not so sure....

I try, but it will get wrong result. so I do not use it.

Reply:
Do you know why "Encode" makes a difference result?

Reply:
I found that following code will get correct result.
```go
    strBytes := buf.GetBytes(i)

    // According to doc: Empty strings are stored as empty strings.
    if len(strBytes) == 0 {
      result.AppendString("")
    }

    strBytes, err = bufEnc.Encode(encodedBuf, strBytes)
    if err != nil {
      return err
    }

    compressed, err := deflate(strBytes)
```
However, this will get wrong result:
```go
    strBytes := buf.GetBytes(i)

    // According to doc: Empty strings are stored as empty strings.
    if len(strBytes) == 0 {
      result.AppendString("")
    }

    strB1, err := bufEnc.Encode(encodedBuf, strBytes)
    if err != nil {
      return err
    }

    compressed, err := deflate(strB1)
```
Quite confused...

Reply:
Great, I fix it.